oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -463,26 +465,15 @@ public static ServiceAccountCredentials fromStream(InputStream credentialsStream
    */
   public static ServiceAccountCredentials fromStream(
       InputStream credentialsStream, HttpTransportFactory transportFactory) throws IOException {
-    Preconditions.checkNotNull(credentialsStream);
-    Preconditions.checkNotNull(transportFactory);
-
-    JsonFactory jsonFactory = OAuth2Utils.JSON_FACTORY;
-    JsonObjectParser parser = new JsonObjectParser(jsonFactory);
-    GenericJson fileContents =
-        parser.parseAndClose(credentialsStream, StandardCharsets.UTF_8, GenericJson.class);
-
-    String fileType = (String) fileContents.get("type");
-    if (fileType == null) {
-      throw new IOException("Error reading credentials from stream, 'type' field not specified.");
-    }
-    if (SERVICE_ACCOUNT_FILE_TYPE.equals(fileType)) {
-      return fromJson(fileContents, transportFactory);
+    ServiceAccountCredentials credential =
+        (ServiceAccountCredentials)
+            GoogleCredentials.fromStream(credentialsStream, transportFactory);
+    if (credential == null) {
```

Comment:
nit: could this ever actually be hit? Wouldn't the fromStream either return a credential (which will throw an error on the cast if its not serviceAccountCredentials) or throw an error?

Reply:
It will be null if the returning type would be different from ServiceAccountCredentials

Reply:
I will add a test for that if there is none yet